### PR TITLE
Feature/pt 1214 crowdin integration

### PR DESCRIPTION
**Pull request by ydrisR**
* Add the the javascript bits required to use the Crowdin live preview features in hydromail.
* Also update the locale files (Crowdin export automatically remove empty lines)

### CODE REVIEW

File: crowdin.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+files:
+  - source: /templates/locales/strings.yaml
```
Comment (BigFab) [inline crowdin.yml:2]: can we use _/templates/locales/en.yaml_ for the source? so we don't have to maintain 2 files for same purpose?

**Review by BigFab — COMMENTED** [review]
_refreshLocales.go_ shall we change line 5 to be:
```go
// Used to interact with a remote localization service such as Crowdin
```
instead of
```go
// Used to interact with a remote localization service such as LoCo or Lokalize
```

File: templates/preview/mailPreview.go [hunk 2]
```diff
@@ -44,7 +44,7 @@ func main() {
 		log.Fatal(err)
 	}
 	router := mux.NewRouter()
-	// localizationManager = NewLocoManager(config.LocalizeServiceUrl, config.LocalizeServiceAuthKey)
+	// localizationManager := NewLokaliseManager(config.LocalizeServiceUrl, config.LocalizeServiceAuthKey)
```
Comment (BigFab) [inline templates/preview/mailPreview.go]:
```suggestion
```

File: templates/preview/local-env.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+export TIDEPOOL_HYDROPHONE_SERVICE="{
```
Comment (BigFab) [inline templates/preview/local-env.sh]: do we need 2 environment files? env.sh and local-env.sh

File: templates/preview/env.sh [hunk 4]
```diff
@@ -2,10 +2,10 @@ export TIDEPOOL_HYDROPHONE_SERVICE="{
...
     \"serverSecret\": \"secret\",
-    \"webUrl\": \"https://android-qa.your-loops.dev\",
+    \"webUrl\": \"https://preview.your-loops.dev\",
     \"patientPasswordResetUrl\": \"https://support.diabeloop.com/hc/articles/360025168234\",
-    \"supportUrl\": \"mailto:yourloops@diabeloop.fr\",
-    \"i18nTemplatesPath\": \"../\",
-    \"localizeServiceUrl\": \"https://localise.biz/api\",
+    \"supportUrl\": \"mailto:yourloops@diabeloop.com\",
+    \"i18nTemplatesPath\": \"./\",
+    \"localizeServiceUrl\": \"xxx\",
```
Comment (BigFab) [inline templates/preview/env.sh]:
```suggestion
```

File: templates/preview/env.sh [hunk 5]
```diff
@@ -2,10 +2,10 @@ export TIDEPOOL_HYDROPHONE_SERVICE="{
...
     \"localizeServiceAuthKey\": \"xxxxxxx\"
```
Comment (BigFab) [inline templates/preview/env.sh]:
```suggestion
```

File: templates/preview/env.sh [hunk 6]
```diff
@@ -2,10 +2,10 @@ export TIDEPOOL_HYDROPHONE_SERVICE="{
...
     \"assetUrl\": \"https://s3-eu-west-1.amazonaws.com/com.diabeloop.public-assets\",
     \"serverSecret\": \"secret\",
-    \"webUrl\": \"https://android-qa.your-loops.dev\",
+    \"webUrl\": \"https://preview.your-loops.dev\",
     \"patientPasswordResetUrl\": \"https://support.diabeloop.com/hc/articles/360025168234\",
-    \"supportUrl\": \"mailto:yourloops@diabeloop.fr\",
-    \"i18nTemplatesPath\": \"../\",
-    \"localizeServiceUrl\": \"https://localise.biz/api\",
+    \"supportUrl\": \"mailto:yourloops@diabeloop.com\",
+    \"i18nTemplatesPath\": \"./\",
```
Comment (BigFab) [inline templates/preview/env.sh]:
```suggestion
    \"i18nTemplatesPath\": \"./\"
```

File: templates/preview/local-env.sh [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+    \"httpaddr\": \"localhost:8088\",
+    \"assetUrl\": \"https://s3-eu-west-1.amazonaws.com/com.diabeloop.public-assets\",
+    \"serverSecret\": \"secret\",
+    \"webUrl\": \"https://preview.your-loops.dev\",
+    \"patientPasswordResetUrl\": \"https://support.diabeloop.com/hc/articles/360025168234\",
+    \"supportUrl\": \"mailto:yourloops@diabeloop.com\",
+    \"i18nTemplatesPath\": \"../\",
```
Comment (BigFab) [inline templates/preview/local-env.sh]:
```suggestion
    \"i18nTemplatesPath\": \"../\"
```

File: templates/preview/local-env.sh [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+    \"localizeServiceUrl\": \"xxx\",
```
Comment (BigFab) [inline templates/preview/local-env.sh]:
```suggestion
```

File: templates/preview/local-env.sh [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+    \"localizeServiceAuthKey\": \"xxxxxxx\"
```
Comment (BigFab) [inline templates/preview/local-env.sh]:
```suggestion
```

File: crowdin.yml (hunk 1 above)
Comment (ydrisR) [reply crowdin.yml:2]: no, I changed this on purpose.
By the way, we don't need to maintain the file en.yaml which will be maintained from crowdin

File: templates/preview/local-env.sh (hunk 3 above)
Comment (ydrisR) [reply templates/preview/local-env.sh]: we need the local-env.sh for local dev
The other environment file contain what is needed when it is deployed. It is not mandatory as we use ansible to generate a configuration file anyway...
Is it a problem to ship the package with a default env file?

File: crowdin.yml [hunk 10]
```diff
@@ -0,0 +1,3 @@
+files:
```
Comment (BigFab) [inline crowdin.yml:1]: I understand this file is to be used for Crowdin CLI. Why do we need it here?

File: templates/preview/local-env.sh (hunk 3 above)
Comment (BigFab) [reply templates/preview/local-env.sh]: I thought we usually have 1 env file that we override based on context (local dev, deployment here or there..)

**Comment by BigFab** [comment]
I finished my review. I still have few questions for you @ydrisR

**Review by BigFab — APPROVED** [review]
LGTM Very good job @ydrisR :+1:

**Pull request merged**